src/TuDa.CIMS.Web/Components/ShoppingCart/ShoppingCartSubmitPopup.razor.cs

```diff
@@ -37,6 +45,21 @@ public partial class ShoppingCartSubmitPopup
     /// <summary>
     /// Closes the MudDialog.
     /// </summary>
-    private void Submit() =>
-        MudDialog.Close(DialogResult.Ok(new WorkingGroupWithBuyer(WorkingGroup!.Id, Buyer.Id)));
+    private void Submit()
+    {
+        if (WorkingGroup?.Students != null)
+        {
+            List<Guid> studentIds = [];
+            studentIds.AddRange(WorkingGroup.Students.Select(student => student.Id));
+            if (WorkingGroup?.Professor.Id != Buyer.Id && !studentIds.Contains(Buyer.Id))
+            {
+                BuyerIsValid = false;
+                BuyerValidationMessage = "Der ausgewählte Käufer gehört nicht zur ausgewählten Arbeitsgruppe.";
+            }
+            if(IsValid)
+            {
+                MudDialog.Close(DialogResult.Ok(new WorkingGroupWithBuyer(WorkingGroup!.Id, Buyer.Id)));
+            }
```

Comment:
⛏️ I would switch it around.
```suggestion
            if (IsValid && (WorkingGroup?.Professor.Id = Buyer.Id || studentIds.Contains(Buyer.Id)))
            {
                 MudDialog.Close(DialogResult.Ok(new WorkingGroupWithBuyer(WorkingGroup!.Id, Buyer.Id)));
            }
            esle
            {
                BuyerIsValid = false;
                BuyerValidationMessage = "Der ausgewählte Käufer gehört nicht zur ausgewählten Arbeitsgruppe.";
            }
```